atlassian_jwt_auth/verifier.py

```diff
@@ -27,25 +29,39 @@ def verify_jwt(self, a_jwt, audience, leeway=0, **requests_kwargs):
         key_identifier = key._get_key_id_from_jwt_header(a_jwt)
         public_key = self.public_key_retriever.retrieve(
             key_identifier, **requests_kwargs)
+
+        # We extract the method to let subclasses convert it to a coroutine
+        return self._decode_jwt(
+            a_jwt, key_identifier.key_id, public_key,
+            audience=audience, options=options, leeway=leeway)
+
+    def _decode_jwt(self, a_jwt, key_id, jwt_key,
+                    audience=None, options=None, leeway=0):
+        """Decode JWT and check if it's valid"""
         claims = jwt.decode(
-            a_jwt, key=public_key,
+            a_jwt,
+            key=jwt_key,
             algorithms=self.algorithms,
             options=options,
             audience=audience,
             leeway=leeway)
-        if not (key_identifier.key_id.startswith('%s/' % claims['iss']) or
-                key_identifier.key_id == claims['iss']):
+
+        if not key_id.startswith('%s/' % claims['iss']) and \
+                key_id != claims['iss']:
             raise ValueError('Issuer does not own the supplied public key')
+
         if self._subject_should_match_issuer and (
                 claims.get('sub') and claims['iss'] != claims['sub']):
             raise ValueError('Issuer does not match the subject.')
+
```

Comment:
It's an advice from [PEP8](https://www.python.org/dev/peps/pep-0008/#blank-lines):

> Use blank lines in functions, sparingly, to indicate logical sections.